spec.md

```diff
@@ -1904,9 +1905,14 @@ message NodePublishVolumeRequest {
   // The path to which the volume will be published. It MUST be an
   // absolute path in the root filesystem of the process serving this
   // request. The CO SHALL ensure uniqueness of target_path per volume.
-  // The CO SHALL ensure that the process serving the request has
-  // `read` and `write` permissions to the path, and is able to create
-  // files or directories at the path if it does not exist.
+  // The CO SHALL ensure that the parent directory of this path exists
+  // and that the process serving the request has `read` and `write`
+  // permissions to that parent directory.
+  // For volumes with an access type of block, the SP SHALL place the
+  // block device at target_path.
+  // For volumes with an access type of mount, the SP SHALL place the
+  // mounted directory at target_path.
+  // Creation of target_path is the responsibility of the SP.
```

Comment:
Nothing needs to be changed here, just leaving a note.

Since the spec never discuss the relationship between volume states (`NODE_READY`, `VOL_READY` and `PUBLISHED`) and node reboots, it could be argued that a `PUBLISHED` volume could be in one of either states after a reboot:
1. `PUBLISHED` since no `NodeUnpublishVolume` has been called. In this case, the CO might call `NodeUnpublishVolume` -> `NodeUnstageVolume` -> `NodeStageVolume` -> `NodePublishVolume` to recover the volume. As a result, the plugin must be able to handle the first two calls when the mount point is missing in the mount table.
2. `NODE_READY` since the mount table has been reset after reboot. In this case, the CO might directly call `NodeStageVolume` and `NodePublishVolume` to recover the volume. As a result, the plugin must be able to deal with the fact that `target_path` might have been created by itself in a previous run.

cc @jdef

Reply:
Thanks for leaving this comment. At some point I think we should consider adding a "Notes for plugin writers" section to capture things along these lines. If people aren't comfortable adding to the spec.md file then we could add a separate "notes-for-plugin-writers.md" file or something that gives additional context, examples, and other things to consider when writing plugins.